Packs/PcapAnalysis/Scripts/PcapFileExtractor/PcapFileExtractor.py

```diff
@@ -82,23 +173,38 @@ def upload_files(dir_path: str, file_path: str) -> Union[CommandResults, str]:
             outputs=context,
             readable_output=readable_output
         )
-
-    return results
-
-
-def main():
-    dir_path = mkdtemp()
-    try:
-        args = demisto.args()
-        file_path = get_pcap_path(args)
-        return_results(upload_files(dir_path, file_path))
-
-    except Exception as e:
-        return_error(f'Failed to execute PcapFileExtracor. Error: {str(e)}')
-    finally:
-        if dir_path:
-            shutil.rmtree(dir_path)
+        return results
+    else:
+        raise DemistoException('No files found in path.')
+
+
+def main(
+        entry_id: str,
+        wpa_password: Optional[str] = None,
+        rsa_decrypt_key_entry_id: Optional[str] = None,
+        types: Optional[str] = None,
+        types_inclusive_or_exclusive: Optional[str] = 'inclusive',
+        extensions: Optional[str] = None,
+        extensions_inclusive_or_exclusive: str = 'inclusive',
+        limit: Optional[str] = '5',
+):
+    with tempfile.TemporaryDirectory() as dir_path:
+        try:
+            file_path, file_name = get_file_path_from_id(entry_id)
+            cert_path, _ = get_file_path_from_id(rsa_decrypt_key_entry_id)
```

Comment:
i think it makes sense to have the condition here to check if `rsa_decrypt_key_entry_id` was provided then get its file path, rather than returning `None, None` from the func